plugins/sources/prometheus/prometheus.go

```diff
@@ -179,93 +179,113 @@ func (src *prometheusMetricsSource) buildPoints(metricFamilies map[string]*dto.M
+		Tags:      tags,
 	}
 }
 
 // Get name and value from metric
-func (src *prometheusMetricsSource) buildPoint(name string, m *dto.Metric, now int64, tags string) []*metrics.MetricPoint {
+func (src *prometheusMetricsSource) buildPoint(name string, m *dto.Metric, now int64, tags map[string]string) []*metrics.MetricPoint {
 	var result []*metrics.MetricPoint
 	if m.Gauge != nil {
 		if !math.IsNaN(m.GetGauge().GetValue()) {
 			point := src.metricPoint(name+".gauge", float64(m.GetGauge().GetValue()), now, src.source, tags)
-			result = src.filterAppend(result, point)
+			result = append(result, point)
 		}
 	} else if m.Counter != nil {
 		if !math.IsNaN(m.GetCounter().GetValue()) {
 			point := src.metricPoint(name+".counter", float64(m.GetCounter().GetValue()), now, src.source, tags)
-			result = src.filterAppend(result, point)
+			result = append(result, point)
 		}
 	} else if m.Untyped != nil {
 		if !math.IsNaN(m.GetUntyped().GetValue()) {
 			point := src.metricPoint(name+".value", float64(m.GetUntyped().GetValue()), now, src.source, tags)
-			result = src.filterAppend(result, point)
+			result = append(result, point)
 		}
 	}
 	return result
 }
 
 // Get Quantiles from summary metric
-func (src *prometheusMetricsSource) buildQuantiles(name string, m *dto.Metric, now int64, tags string) []*metrics.MetricPoint {
+func (src *prometheusMetricsSource) buildQuantiles(name string, m *dto.Metric, now int64, tags map[string]string) []*metrics.MetricPoint {
 	var result []*metrics.MetricPoint
 	for _, q := range m.GetSummary().Quantile {
 		if !math.IsNaN(q.GetValue()) {
-			newTags := fmt.Sprintf("%s quantile=%v", tags, q.GetQuantile())
-			point := src.metricPoint(name, float64(q.GetValue()), now, src.source, newTags)
-			result = src.filterAppend(result, point)
+			tags["quantile"] = fmt.Sprintf("%v", q.GetQuantile())
+			point := src.metricPoint(name, float64(q.GetValue()), now, src.source, tags)
+			result = append(result, point)
 		}
 	}
 	point := src.metricPoint(name+".count", float64(m.GetSummary().GetSampleCount()), now, src.source, tags)
-	result = src.filterAppend(result, point)
+	result = append(result, point)
 	point = src.metricPoint(name+".sum", float64(m.GetSummary().GetSampleSum()), now, src.source, tags)
-	result = src.filterAppend(result, point)
+	result = append(result, point)
 
 	return result
 }
 
 // Get Buckets from histogram metric
-func (src *prometheusMetricsSource) buildHistos(name string, m *dto.Metric, now int64, tags string) []*metrics.MetricPoint {
+func (src *prometheusMetricsSource) buildHistos(name string, m *dto.Metric, now int64, tags map[string]string) []*metrics.MetricPoint {
 	var result []*metrics.MetricPoint
 	for _, b := range m.GetHistogram().Bucket {
-		newTags := fmt.Sprintf("%s le=%v", tags, b.GetUpperBound())
-		point := src.metricPoint(name, float64(b.GetCumulativeCount()), now, src.source, newTags)
-		result = src.filterAppend(result, point)
+		tags["bucket"] = fmt.Sprintf("%v", b.GetUpperBound())
```

Comment:
change `bucket` to `le`. I had done that change recently since that's how prometheus represents it.